pkg/services/query/query.go

```diff
@@ -90,12 +91,20 @@ func (s *Service) QueryData(ctx context.Context, user *user.SignedInUser, skipCa
 	g, ctx := errgroup.WithContext(ctx)
 	results := make([]backend.Responses, len(byDataSource))
 
+	// Create panic recovery function for loop
+	recoveryFn := func() {
+		if e := recover(); e != nil {
+			s.log.Error("query datasource panic", "error", e, "stack", string(debug.Stack()))
```

Comment:
Might want to use something like this and make sure to populate a data response, if possible, to make sure the full response includes results of all the queries:
https://github.com/grafana/grafana/blob/b799be3052247f6d3da2a10ab18474dcb0705a5d/pkg/tsdb/sqleng/sql_engine.go#L219-L227

The pkg/infra/log package is used in a 4 different places where `debug.Stack` used in 2 places. I don't know which one is the preferred. Maybe @grafana/backend-platform can bring some clarity?

Reply:
done in the recent commit